src/main/daml/Daml/Finance/Settlement/Batch.daml

```diff
@@ -57,28 +61,26 @@ template Batch
         let
           settleInstruction instructionKey = do
             let context = show instructionKey
+            -- execute instruction and get used token standards for the instrument
             (instructionCid, instruction) <- fetchByKey @Instruction instructionKey
-            pledged <- case instruction.allocation of
-              Pledge cid -> Some <$> fetch cid
-              _ -> pure None
+            pledgedTokenStandard <- case instruction.allocation of
+              Pledge cid -> L.singleton <$> getTokenStandard cid
+              _ -> pure []
             settledCid <- exercise (toInterfaceContractId @Instruction.I instructionCid)
               Instruction.Execute with actors = actors `S.union` requestors
-            settled <- T.mapA fetch settledCid
+            settledTokenStandard <- optionalToList <$> T.mapA getTokenStandard settledCid
+
             let
-              usedInterfaceTypeReps =
-                fmap ((instruction.routedStep.quantity.unit, ) . interfaceTypeRep) <$>
-                [pledged, settled]
-            settledCid <-
-              T.mapA
-                (\cid -> fromOptional cid <$> undisclose (context, settlers) actors cid)
-                settledCid
-            pure (settledCid, usedInterfaceTypeReps)
+              instrumentWithTokenStandards = fmap ((instruction.routedStep.quantity.unit, )) $
```

Comment:
do you need the double bracket?

Reply:
no, changed it to `(instruction.routedStep.quantity.unit, ) <$> settledHoldingStandard <> pledgedHoldingStandard`